src/lualib/ArraySetLength.ts

```diff
@@ -0,0 +1,15 @@
+function __TS__ArraySetLength<T>(this: void, arr: T[], length: number): number {
+    if (length < 0
+        || length !== length // NaN
+        || length === (1 / 0) // Infinity
+        || length === (-1 / 0) // -Infinity
```

Comment:
I think `length < 0` should be enough to catch it

Reply:
good point